src/MoBi.Presentation/Presenter/Main/ModuleExplorerPresenter.cs

```diff
@@ -0,0 +1,168 @@
+
+         return base.ContextMenuFor(treeNode);
+      }
+
+      public override void NodeDoubleClicked(ITreeNode node)
+      {
+         var moleculeBuilder = node.TagAsObject as IMoleculeBuilder;
+         if (moleculeBuilder == null)
+         {
+            base.NodeDoubleClicked(node);
+            return;
+         }
+
+         var moleculeBuildingBlock = node.ParentNode.TagAsObject.DowncastTo<IMoleculeBuildingBlock>();
+         _editBuildingBlockStarter.EditMolecule(moleculeBuildingBlock, moleculeBuilder);
+      }
+
+      public bool ShouldSort(ITreeNode node)
+      {
+         if (node.ParentNode == null)
+            return false;
+
+         // Do not sort if this is content of a module.
+         // Child nodes are created in the same order as the older Building Block Explorer
+         return !node.ParentNode.TagAsObject.IsAnImplementationOf<Module>();
+      }
+
+      public override bool CanDrag(ITreeNode node)
+      {
+         if (node == null)
+            return false;
+
+         if (node.IsAnImplementationOf<ClassificationNode>())
+            return true;
+
+         return _observedDataInExplorerPresenter.CanDrag(node);
+      }
+
+      public override IEnumerable<ClassificationTemplate> AvailableClassificationCategories(ITreeNode<IClassification> parentClassificationNode)
+      {
+         return _observedDataInExplorerPresenter.AvailableObservedDataCategoriesIn(parentClassificationNode);
+      }
+
+      public override void AddToClassificationTree(ITreeNode<IClassification> parentNode, string category)
+      {
+         _observedDataInExplorerPresenter.GroupObservedDataByCategory(parentNode, category);
+      }
+
+      public override bool RemoveDataUnderClassification(ITreeNode<IClassification> classificationNode)
+      {
+         return _observedDataInExplorerPresenter.RemoveObservedDataUnder(classificationNode);
+      }
+
+      private void addBuildingBlockToTree<TBuildingBlock>(TBuildingBlock buildingBlock, RootNodeType buildingBlockFolderType) where TBuildingBlock : IBuildingBlock
+      {
+         var buildingBockFolderNode = _view.NodeByType(buildingBlockFolderType);
+
+         addBuildingBlockUnderNode(buildingBlock, buildingBockFolderNode);
+      }
+
+      private void addBuildingBlockUnderNode<TBuildingBlock>(TBuildingBlock buildingBlock, ITreeNode folderNode) where TBuildingBlock : IBuildingBlock
+      {
+         if (buildingBlock == null)
+            return;
+
+         _view.AddNode(_treeNodeFactory.CreateFor(buildingBlock)
+            .Under(folderNode));
+      }
+
+      protected override void AddProjectToTree(IMoBiProject project)
+      {
+         using (new BatchUpdate(_view))
+         {
+            _view.DestroyNodes();
+
+            project.Modules.Each(addModule);
+
+            _view.AddNode(_treeNodeFactory.CreateFor(MoBiRootNodeTypes.ExpressionProfilesFolder));
+            _view.AddNode(_treeNodeFactory.CreateFor(MoBiRootNodeTypes.IndividualsFolder));
```

Comment:
why that?  Shouldn't that be outside of the module explorer?

Reply:
Do we have a separate explorer view for ObservedData, Individuals and Expressions? I didn't think so. They appear in this view, but are not part of a module domain object.

That was my understanding.